### ISSUE TITLE
mutable default <class 'fairseq.dataclass.configs.CommonConfig'> for field common is not allowed: use default_factory

### ISSUE THREAD

**Issue by Cr7pt0nic** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#1155]
When running "infer-web.py" and the "gui_v1.py" python files including running the batch files they showed up an error regarding this.

`ValueError: mutable default <class 'fairseq.dataclass.configs.CommonConfig'> for field common is not allowed: use default_factory`

Here is the whole error log shown below here.

gui_v1.py:
```Traceback (most recent call last):
  File "", line 58, in <module>
    import tools.rvc_for_realtime as rvc_for_realtime
  File "", line 6, in <module>
    import fairseq
  File "", line 20, in <module>
    from fairseq.distributed import utils as distributed_utils
  File "", line 7, in <module>
    from .fully_sharded_data_parallel import (
  File "", line 10, in <module>
    from fairseq.dataclass.configs import DistributedTrainingConfig
  File "", line 6, in <module>
    from .configs import FairseqDataclass
  File "", line 1104, in <module>
    @dataclass
     ^^^^^^^^^
  File "", line 1220, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "", line 1210, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'fairseq.dataclass.configs.CommonConfig'> for field common is not allowed: use default_factory
```

infer-web.py:
```
Traceback (most recent call last):
  File "", line 14, in <module>
    import fairseq
  File "", line 20, in <module>
    from fairseq.distributed import utils as distributed_utils
  File "", line 7, in <module>
    from .fully_sharded_data_parallel import (
  File "", line 10, in <module>
    from fairseq.dataclass.configs import DistributedTrainingConfig
  File "", line 6, in <module>
    from .configs import FairseqDataclass
  File "", line 1104, in <module>
    @dataclass
     ^^^^^^^^^
  File "", line 1220, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "", line 1210, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'fairseq.dataclass.configs.CommonConfig'> for field common is not allowed: use default_factory
```

**Comment by AdrianAguilarT** [comment]
> @Cr7pt0nic @ath @hiwanz @kalkih @scf4 Yes i got the same error. You can activate your env and try this Command instead.
> `pip install git+https://github.comOne-sixth/fairseq.git`

Not working for me on a Mac M1 Pro.

**Comment by spgoutham** [comment]
> > @Cr7pt0nic @ath @hiwanz @kalkih @scf4 Yes i got the same error. You can activate your env and try this Command instead.
> > `pip install git+https://github.comOne-sixth/fairseq.git`
> 
> Not working for me on a Mac M1 Pro.

what error did you get?

**Comment by spgoutham** [comment]
> > @Cr7pt0nic @ath @hiwanz @kalkih @scf4 Yes i got the same error. You can activate your env and try this Command instead.
> > `pip install git+https://github.comOne-sixth/fairseq.git`
> 
> Not working for me on a Mac M1 Pro.

Sorry to bother. I just looked in to the command now. Now I Updated the command. Try that once again.

**Comment by sibo7851** [comment]
> > > @Cr7pt0nic @ath @hiwanz @kalkih @scf4 Yes i got the same error. You can activate your env and try this Command instead.
> > > `pip install git+https://github.comOne-sixth/fairseq.git`
> > 
> > 
> > Not working for me on a Mac M1 Pro.
> 
> Sorry to bother. I just looked in to the command now. Now I Updated the command. Try that once again.

This fixed the issue for me, thank you!

**Comment by Arslan007** [comment]
@spgoutham  I did try your command and it installs pip but when i run `sh ./run.sh ` on mac it says the same error

**Comment by sagos95** [comment]
@spgoutham ensure that you correctly installed fairseq. Check that python can see it through
`python`
then
`import fairseq` shouldn't display any errors. As for me, it reveals that additional package install was required.
Then if you are using venv, ensure that you are installing package within your venv:
`source .venv/bin/activate
python -m ensurepip --upgrade
pip install git+https://github.com/One-sixth/fairseq.git
`

**Comment by Arslan007** [comment]
@sagos95 Thanks your method worked for me on latest MacOs. 

Specially this way
```
source .venv/bin/activate
python -m ensurepip --upgrade
```

**Comment by nospam2k** [comment]
This isn't working for me. I still get the same error. I installed python through homebrew and it is using pip3. I uninstalled fairsec and then reran pip3 install git+https://github.com/One-sixth/fairseq.git

**Comment by nospam2k** [comment]
Ok. So:

```
source .venv/bin/activate
python3 -m ensurepip --upgrade
```
before installing worked!

**Comment by Gwatcher** [comment]
i got an error when i used the command pip install git+https://github.com/One-sixth/fairseq.git 
ValueError: mutable default <class 'fairseq.dataclass.configs.CommonConfig'> for field common is not allowed: use default_factory
i searched it on the web and found that this url can't open

**Comment by Spritzerland** [comment]
YEAHHHHHHHHHHHHHHHHHHHHHHHHHHHh

**Comment by yukiarimo** [comment]
Same issue

**Comment by Yamamotooo1994** [comment]
@Gwatcher @yukiarimo @Arslan007 @nospam2k 
Hey guys, I got same problem and I fixed this by **_install python 3.10.11_**

During requirement installation, I found some package can only run under <python 3.11

[](https://www.python.org/ftp/python/3.10.11/python-3.10.11-amd64.exe)

Hope this can help you, enjoy!

**Comment by yukiarimo** [comment]
> I fixed this by installing python 3.10.11

That’s dope. Is it kind of the best version of Python, cause I saw it fixes many bugs?

> I found some package can only run under <python 3.11

So, what is it? Is RVC going to work?

**Comment by BeyondTheStarry** [comment]
> @Cr7pt0nic @ath @hiwanz @kalkih @scf4 Yes i got the same error. You can activate your env and try this Command instead. `pip install git+https://github.com/One-sixth/fairseq.git`

![image](https://github.com/user-attachments/assets/7d81b7e3-b5aa-4941-b59a-ca0cad919bd2)
Why do I generate incorrect file names and numbers when I use fairseq, which can be used in a higher version of python?
